core/src/main/java/bio/terra/pearl/core/service/datarepo/DataRepoExportService.java

```diff
@@ -167,14 +170,14 @@ public void ingestDataForStudyEnvironment(Dataset studyEnvDataset) {
         }
     }
 
-    public List<String> generateDatasetSchema(UUID studyEnvironmentId) {
+    public Map<String, TableDataType> generateDatasetSchema(UUID studyEnvironmentId) {
         ExportOptions exportOptions = new ExportOptions(false, false, false, ExportFileFormat.TSV, null);
 
         //Backtrack from studyEnvironmentId to get the portalId, so we can export the study environment data
         StudyEnvironment studyEnv = studyEnvironmentDao.find(studyEnvironmentId).orElseThrow(() -> new NotFoundException("Study environment not found."));
         PortalStudy portalStudy = portalStudyDao.findByStudyId(studyEnv.getStudyId()).stream().findFirst().orElseThrow(() -> new NotFoundException("Portal study not found."));
 
-        List<String> columnKeys = new ArrayList<>();
+        Map<String, TableDataType> columnKeys = new LinkedHashMap<>();
```

Comment:
I concur-- instead of a Map, this should be a List of "TdrColumn" objects, and for now we can define "TdrColumn" as a record of name and datatype, but I wouldn't be surprised if we add more later.

Reply:
For me, this isn't about ordering (it's usually bad practice to rely on column ordering in any external database-like-system), and more about readability.  (Which might be an argument that it should be a Set rather than a List, but I'll defer to your TDR expertise on which is best)

Reply:
I added two new record types: `TdrColumn` and `TdrTable`, and switched these over to sets. But I've kept it as a `LinkedHashSet` because I think the ordering is important. The column order in TDR seems to be dependent on the order in which the columns were initialized. So maintaining the order in our code allows us to put things like `enrollee_shortcode` and basic profile information first in the TDR UI, and it also keeps particular survey questions grouped together. Otherwise you'll just see a bunch of columns like `oh_oh_medHx.otherKidneyConditionAgeAtDx` at the beginning of the dataset which I don't think is a great UX when viewing the dataset.

I've added a code comment to explain this, but am happy to revisit the decision.